Feature/professor lol 45

jyami-kim opened the pull request.

Commit:
도메인 구성:

Commit:
professor-lol-45 패치 히스토리 도메인 구현

도메인 테스트 구현중

jyami-kim committed:
@DataJPA Test 완료

csbsjy commented on professorlol-core/src/main/java/com/ccs/professorlol/patch/champion/PatchHistory.java:
```diff
@@ -0,0 +1,33 @@
+    @JoinColumn(name = "lol_info_id")
+    private LolInfo lolInfo;
+
+    private String patchVersion;
```
이 patchVersion이랑 LolInfo 안의 patchNoteVersion이랑 다른거ㅇ야?

chldbtjd2272 commented on professorlol-core/src/main/java/com/ccs/professorlol/patch/skill/NewAbility.java:
```diff
@@ -0,0 +1,23 @@
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@DiscriminatorValue("2")
```
직관적인 이름으로 변경

Gobukgol commented on professorlol-web/src/main/java/com/ccs/professorlol/service/PatchHistoryValidator.java:
```diff
@@ -0,0 +1,27 @@
+@Component
+@RequiredArgsConstructor
+public class PatchHistoryValidator<T> {
+    private final PatchHistoryRepository patchHistoryRepository;
```
이 친구는 어디에 쓰이는거야?

jyami-kim replied:
나중에 추가하려고 만든거 같은데 까먹었나보다ㅋㅋ

Gobukgol commented on professorlol-web/src/main/java/com/ccs/professorlol/service/PatchHistoryService.java:
```diff
@@ -0,0 +1,38 @@
+
+    }
+
+    private List<LolInfo> findAll(List<Champion> championDtos) {
```
Champion 리스트이니까 champions가 맞지않아?

jyami-kim replied:
엥 그러게 ㄳㄳ

jyami-kim committed:
jpa refactoring

jyami-kim committed:
professor-lol-45 패치 히스토리 도메인 구현

1. DataJPATest - 엔티티 생성 테스트
2. JPA 상속 매핑
3. application.yml 설정 : JPA DB 설정 ( local, test : h2 )

pci2676 commented on professorlol-core/src/main/java/com/ccs/professorlol/lolInfo/domain/MockInfo.java:
```diff
@@ -0,0 +1,12 @@
+@Entity
+@Getter
+@NoArgsConstructor
+public class MockInfo extends LolInfo {
```
test에 사용될 mock객체야? 그러면 test 패키지에 있는게 낫지않을까?

pci2676 commented on professorlol-core/src/main/java/com/ccs/professorlol/patch/champion/ChampionPatchHistory.java:
```diff
@@ -0,0 +1,41 @@
+        this.context = context;
+        this.championAbilityHistories = championAbilityHistories.stream()
+                .map(championAbilityHistory -> {
+                    championAbilityHistory.setChampionPatchHistory(this);
```
이 부분 함수로 추출해주라

pci2676 commented on professorlol-core/src/main/java/com/ccs/professorlol/patch/skill/ChampionAbilityHistory.java:
```diff
@@ -0,0 +1,33 @@
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "patch_history_id")
+    @Setter
```
setter꼭 써야돼??
set하는 기능이 필요하면 명시적인 메소드 이름으로 바꿔줄수있어?

pci2676 commented on professorlol-core/src/test/java/com/ccs/professorlol/patch/champion/PatchHistoryTest.java:
```diff
@@ -0,0 +1,132 @@
+@RunWith(SpringRunner.class)
+@DataJpaTest    //spring context 안띄고 test!
+public class PatchHistoryTest {
+    static String championName = "라이즈";
```
private 일관성있게 다붙여주라

pci2676 commented on professorlol-riot-api/src/main/java/com/ccs/professorlol/LeagueRestTemplateConfig.java:
```diff
@@ -26,12 +29,11 @@
         public LeagueRestTemplate leagueRestTemplate() {
             log.debug("League RestTemplate Builder created");
-            RestTemplateBuilder restTemplateBuilder = RiotRestTemplateBuilder.get(this.restTemplateBuilder, this.riotProperties);
+            RestTemplateBuilder restTemplateBuilder = RiotRestTemplateBuilder.get(new RestTemplateBuilder(), this.riotProperties);
```
이것이었군..!!

Gobukgol commented on professorlol-core/src/test/java/com/ccs/professorlol/patch/champion/ChampionPatchHistoryTest.java:
```diff
@@ -0,0 +1,146 @@
+
+
+    @Test
+    public void 패치_히스토리가_내클래스로저장시_ChampionPatchHistory_테이블로__만들어지는지_확인() {
```
내클래스말구 자식클래스라고 하면 더 알아보기 쉬울거같아!

Gobukgol commented on professorlol-core/src/test/java/com/ccs/professorlol/patch/skill/ChampionAbilityHistoryTest.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    @Test
+    public void championAbilityHistory로_검색시_casting_되는지_확인() {
```
~되는지 확인보다는 성공, 실패 이런식으로 네이밍하는건 어때?

Commit:
professor-lol-45 패치 히스토리 도메인 구현

1. DataJpaTest 트랜잭션 제거
2. enable_lazy_load_no_trans 주석처리

Commit:
riot api jar설정 추가
